jenkinsapi/api.py

```diff
@@ -244,3 +245,14 @@ def search_artifact_by_regexp(jenkinsurl, jobid, artifactRegExp,
                 return art
 
     raise ArtifactsMissing()
+
+
+def toggle_keep_build_forever(jenkinsurl, jobname, build_no,
+                              username=None, password=None, ssl_verify=True):
+    """
+    Toggles the keep this build forever flag for the specified build
+    """
+    crumb = CrumbRequester(username, password, baseurl=jenkinsurl,
```

Comment:
Sorry, but not going to accept this change. This will not work in Jenkins where CSRF protection is switched off. And this code needs to go into `Build` object.